extensions/interactions/RatioExpressionInput/directives/oppia-interactive-ratio-expression-input.component.ts

```diff
@@ -18,111 +18,134 @@
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
 
-require('domain/objects/objects-domain.constants.ajs.ts');
+import { CurrentInteractionService } from 'pages/exploration-player-page/services/current-interaction.service';
+import { InteractionAttributesExtractorService } from 'interactions/interaction-attributes-extractor.service';
+import { InteractionRulesService } from 'pages/exploration-player-page/services/answer-classification.service';
+import { RatioExpressionInputRulesService } from './ratio-expression-input-rules.service';
+import { FocusManagerService } from 'services/stateful/focus-manager.service';
 
-angular.module('oppia').component('oppiaInteractiveRatioExpressionInput', {
-  template: require('./ratio-expression-input-interaction.component.html'),
-  bindings: {
-    savedSolution: '<'
-  },
-  controller: [
-    '$attrs', '$scope', 'CurrentInteractionService',
-    'InteractionAttributesExtractorService',
-    'RatioExpressionInputRulesService',
-    function(
-        $attrs, $scope, CurrentInteractionService,
-        InteractionAttributesExtractorService,
-        RatioExpressionInputRulesService) {
-      var ctrl = this;
-      var errorMessage = '';
-      // Label for errors caused whilst parsing ratio expression.
-      var FORM_ERROR_TYPE = 'RATIO_EXPRESSION_INPUT_FORMAT_ERROR';
-      ctrl.getWarningText = function() {
-        return errorMessage;
-      };
+import { RatioExpressionInputCustomizationArgs } from 'interactions/customization-args-defs';
+import { RatioInputAnswer, InteractionAnswer } from 'interactions/answer-defs';
 
-      ctrl.submitAnswer = function(answer) {
-        try {
-          var ratioExpression =
-            Ratio.fromRawInputString(answer);
-          if (
-            (
-              ratioExpression.getNumberOfTerms() !==
-              ctrl.expectedNumberOfTerms) && (ctrl.expectedNumberOfTerms !== 0)
-          ) {
-            throw new Error(
-              'The creator has specified the number of terms in' +
-              ' the answer to be ' + ctrl.expectedNumberOfTerms + '.');
-          }
-          errorMessage = '';
-          ctrl.RatioExpressionInputForm.answer.$setValidity(
-            FORM_ERROR_TYPE, true);
-          CurrentInteractionService.onSubmit(
-            ratioExpression.getComponents(),
-            RatioExpressionInputRulesService);
-        } catch (parsingError) {
-          errorMessage = parsingError.message;
-          ctrl.RatioExpressionInputForm.answer.$setValidity(
-            FORM_ERROR_TYPE, false);
-        }
-      };
+@Component({
+  selector: 'oppia-interactive-ratio-expression-input',
+  templateUrl: './ratio-expression-input-interaction.component.html',
+  styleUrls: []
+})
+export class InteractiveRatioExpressionInputComponent
+  implements OnInit, OnDestroy {
+  @Input() placeholderWithValue: string = '';
+  @Input() numberOfTermsWithValue: string = '';
+  @Input() labelForFocusTarget: string;
```

Comment:
Do you need to initialize these vars with empty strings if you're already mentioning the type? If yes, do it for the other vars too.

Reply:
Done, no need, just add while testing in local. Remove now.